src/bot-command-handlers.js

```diff
@@ -41,36 +28,177 @@ module.exports = {
     },
 };
 
-function rollDice(args) {
-    let rolls = {};
-    let specs = args.message.split(/[ ,]+/);  // split by comma or whitespace
-
-    if (args.message !== "") {
-        // Loop through the dice specification
-        for (let i = 0; i < specs.length; i++) {
-            let total = 0;
-            let internalRolls = [];
-            let parts = specs[i].split(/[dD]+/);
-            let numDie = parts[0] ? parts[0] : "1";
-            let dieSize = parts[1];
-
-            for (let j = 0; j < numDie; j++) {
-                if (dieSize < 1) {
-                    internalRolls.push(0);
-                } else if (dieSize < 2) {
-                    internalRolls.push(1);
-                    total += 1;
-                } else {
-                    let roll = Math.floor(Math.random() * dieSize) + 1;
-                    internalRolls.push(roll);
-                    total += roll;
-                }
-            }
-
-            rolls[specs[i]] = internalRolls;
+
+function createRollMessages(args){
+
+    let messages = [];
+
+    let separateDieRolls = args.message.split(/[ ,]+/); // split by comma or whitespace
+
+    // create a default specification roll if nothing sent to us
+
+    if (separateDieRolls.length === 0) {
+        let spec = rollSpecification('');
+        let rollResults = rollDice(spec['numDie'], spec['dieSize']);
+        let message = rollResultMessage(spec, rolllResults);
```

Comment:
@sososuaveh Is this variable correct? I see `rolllResult` but the line above you have `rollResults`.